### DISCUSSION TITLE
Idea: Linux backend to compare files on linux systems, Controlled by existing Win GUI

### DISCUSSION THREAD

**Discussion by ph1248** [WinMerge/winmerge#1901] [Ideas]
Hi, 

first of all, many thanks for this great tool. It's one of the best tools I have ever used, FOSS or commercial. Indeed, this is also the reason for this Idea or feature request: I simply haven't found anything as good for linux. 

My use case: 
I want to compare / merge folders or files on linux servers, mostly headless servers. I could of course bind the relevant folders on these servers to drives on my local windows machine via samba or NFS. But this takes time to set up and, more important, this means that all files have to be send to my local machine to be compared, or to be copied, or similar transactions. 

Desired target scenario:
I have no idea how the architecture of winmerge is set up. But I imagine that there is some separation between the GUI and the actual data comparison machine which is controlled by the GUI. In my desired target scenario this data comparison machine is recreated for Linux OSs. This "back end" could be installed on the linux server. It would be controlled via SSH by the existing Winmerge GUI remotely from a windows machine. That means, all commands for data transactions would be send via SSH to the "back end" on the linux machine, the back end would execute the file operations and send the results back to the GUI via SSH. This would be the case for comparison, file copy, etc.

We could have such a remote connection to a winmerge back end for each tab in the winmerge gui. Actually, If you go down that road, this remote control feature could eventually also implemented for windows. 

What do you developers think? Is this possible? 

Best regards, 
Philipp

**Comment by ph1248** [comment]
No comments? Well, it was just a thought, and maybe this enhancement would just cause too much work. But I would be glad if you could keep it in mind, maybe one day someone can start building on it. 

cheers, 
Philipp

**Reply by sdottaka** [reply]
WinMerge is heavily dependent on Windows, so porting the backend to Linux will take a lot of time.

I haven't tried it, but I think it's possible to treat a Linux file system as a drive in Windows and compare the drives with WinMerge, as shown below.
https://github.com/winfsp/sshfs-win
